src/arrayMethodJoin.js

```diff
@@ -4,8 +4,30 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    const newArr = [...this];
+    let sep = separator;
+    let result = '';
+
+    if (sep === null) {
+      sep = 'null';
+    }
+
+    for (let i = 0; i < newArr.length; i++) {
+      let tempEl = newArr[i];
+
+      if (tempEl === null || tempEl === undefined) {
+        tempEl = '';
+      }
```

Comment:
invert this condition, so all you do is
```suggestion
      if (tempEl !== null || tempEl !== undefined) {
        result += tempEl
      }
```